Docs sso tutorial auth0 14185 (#14206)
metabase/metabase: docs/enterprise-guide/saml-auth0.md

![Auth0 Application Addons](images/saml-auth0/auth0saml2addon.png)

### Mapping fields from users in Auth0 to Metabase

On the top of the the **Settings** section on this page, you need to re-enter the value you included in **Allowed Callback URLs** in the field named **Application Callback URL**. Copy and paste the following JSON file in the settings box so that Auth0 can send the correct mappings to Metabase when a user logs in:

```
{
    "mappings": {
        "email":"http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress",
        "given_name":"http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname",
        "family_name":"http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname",
        "groups":"http://schemas.xmlsoap.org/claims/Group"
    }
}
```

If you need to set additional user properties, you can always return to this step to add additional fields to the mappings object.

![Auth0 SAML Mappings](images/saml-auth0/auth0samlmappings.png)

### Configuring Metabase
Next, in the Auth0 Addon: SAML2 Web App popup, click on the **Usage** tab and then configure in Metabase (column on the left) the values that this tab provides (column on the right)

> [!Auth0 SAML usage](images/saml-auth0/auth0samlusage.png)

Reviewer: ```suggestion
![Auth0 SAML usage](images/saml-auth0/auth0samlusage.png)
```
Image link wasn't rendering.

---
| Metabase | Auth0 |
| --- | --- |
| SAML Identity Provider URL | Identity Provider Login URL |
| SAML Identity Provider Issuer | Issuer |
| SAML Identity Provider Certificate | Open the Identity Provider Metadata link and copy the string under the X509Certificate tag

> ---

Reviewer: The formatting of this table is off. Jekyll doesn't present it as a table.

The SAML Application Name" value can be left as the default one (Metabase).

> Save your settings. Then enable SAML(authenticating-with-saml.html#turning-on-saml-based-sso) in Metabase, and you should be good to go!

Reviewer: ```suggestion
Save your settings. Then [enable SAML](authenticating-with-saml.md) in Metabase, and you should be good to go!
```
The link wasn't formatted correctly, and the build process should replace the ".md" with ".html".

# Setting up SAML with Auth0

> Setting up Metabase with Auth0 is extremely easy. Just follow this configuration guide and you'll be ready to go in a few couple of minutes.

Reviewer: ```suggestion
```

Avoid saying anything is easy in docs. Because people might run into situations that make it more complicated, or they have trouble following the guide, or whatever, and it's not easy for them. They get frustrated and we lose trust.

Reviewer 2: got it!

For starting the configuration in Metabase, please follow the guide for [authenticating with SAML](authenticating-with-saml.html)

## Steps

## Create an Application

> After you log in to your Auth0 account, navigate to **Applications** and select **Create Application*

Reviewer: ```suggestion
After you log in to your Auth0 account, navigate to **Applications** and select **Create Application**.
```

```
{
    "mappings": {
        "email":"http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress",
        "given_name":"http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname",
        "family_name":"http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname",
        "groups":"http://schemas.xmlsoap.org/claims/Group"
    }
}
```

If you need to send more data from the user properties later, you can always return to this step and see which field you want to map and add.

![Auth0 SAML Mappings](images/saml-auth0/auth0samlmappings.png)

### Configuring Metabase
Next, in the Auth0 Addon: SAML2 Web App popup, click on the **Usage** tab and then configure in Metabase (column on the left) the values that this tab provides (column on the right)

[!Auth0 SAML usage](images/saml-auth0/auth0samlusage.png)

---
| Metabase | Auth0 |
| --- | --- |
| SAML Identity Provider URL | Identity Provider Login URL |
| SAML Identity Provider Issuer | Issuer |
| SAML Identity Provider Certificate | Open the Identity Provider Metadata link and copy the string under the X509Certificate tag
---

The "SAML Application Name" value can be left as the default one (Metabase)

> Click in Save and enable SAML in Metabase with the toggle in the top of the page and you are good to go!

Reviewer 2: just changed the URL to html instead of md (don't know if this is updated automatically when this ships)
